code/drasil-printers/lib/Language/Drasil/TeX/Print.hs

```diff
@@ -76,10 +76,22 @@ lo (Graph ps w h c l)    _  = toText $ makeGraph
 lo (Cell _) _               = empty
 lo (CodeBlock _) _          = empty
 
+-- | Helper for converting layout objects into a more printable form.
+-- This function is specific to definitions.
+lo' :: LayoutObj -> PrintingInformation -> D
+lo' (HDiv _ con _)      sm = printDef sm con
```

Comment:
why does this dispatch to `printDef` while below it goes to `lo` ?

Reply:
First, I'll provide what the difference is between `lo` and `lo'`.

`lo` is for printing `LayoutObj`s normally (i.e. anything that is not inside of a definition table).
`lo'` is for printing `LayoutObj`s that are inside a definition table.

This distinction had to be made because `EqnBlock`s that are not in a definition table can span the entire width of the page (i.e. their `scale` parameter is `1`). Whereas `EqnBlock`s inside definition tables are restricted to 0.8 * page width (i.e. their `scale` parameter is `0.8`)

```Latex
\resizeExpression{...}{1.0} % Outside definition table
\resizeExpression{...}{0.8} % Inside definition table
```

---

The reason the `HDiv` pattern match dispatches to `printDef` (i.e. uses `lo'`) is because `con` is of type `[LayoutObj]`, thus those could also contain `EqnBlock`s. 

The last pattern match is a "catch all" statement, where if the `LayoutObj` is not an `EqnBlock`, it will print the same as `lo`.

Reply:
Very thorough explanation, thank you.

Having `lo`/`lo'` will do. The right solution would be to make our printers use an environment that could carry context (i.e. inside of definition table or not) instead of doing that via different functions.

For the scale parameter, I think it would be good to generate cleaner LaTeX, i.e. generate two variable definitions (at the 'top' of the LaTeX file) for each of the scales, and then use those variables instead of hard-coded numbers. LaTeX is a programming language, after all, and we should use it with the same care as we do other things.

Reply:
Added in https://github.com/JacquesCarette/Drasil/pull/3910/commits/820619edf8387cfd0ecf5a9477cb19232f4aa403 and https://github.com/JacquesCarette/Drasil/pull/3910/commits/845174b40bbf98016564d9fe30f2a4bb0c866ad6.

The following will be defined in the preamble:

```LaTeX
\def\inDefScale{0.8}
\def\outDefScale{1.0}
```

`\resizeExpression` will be used as follows:

```Latex
\resizeExpression{...}{\outDefScale} % Outside definition table
\resizeExpression{...}{\inDefScale} % Inside definition table
```